presto-elasticsearch/src/main/java/com/facebook/presto/elasticsearch/ElasticsearchClient.java

```diff
@@ -0,0 +1,391 @@
+        if (table == null) {
+            return null;
+        }
+        if (table.getColumns().isPresent()) {
+            return table;
+        }
+        List<ElasticsearchColumn> columns = buildColumns(table);
+        ElasticsearchTableDescription result = new ElasticsearchTableDescription(
+                table.getTableName(),
+                table.getSchemaName(),
+                table.getHost(),
+                table.getPort(),
+                table.getClusterName(),
+                table.getIndex(),
+                table.getIndexExactMatch(),
+                table.getType(),
+                Optional.of(columns));
+        return result;
+    }
+
+    public List<String> getIndices(ElasticsearchTableDescription tableDescription)
+    {
+        if (tableDescription.getIndexExactMatch()) {
+            return ImmutableList.of(tableDescription.getIndex());
+        }
+        TransportClient client = clients.get(tableDescription.getClusterName());
+        verify(client != null, "TransportClient is null");
+        String[] indices = client.admin()
+                .indices()
+                .getIndex(new GetIndexRequest())
+                .actionGet(timeout.toMillis())
+                .getIndices();
+        return Arrays.stream(indices)
+                    .filter(index -> index.startsWith(tableDescription.getIndex()))
+                    .collect(toImmutableList());
+    }
+
+    public ClusterSearchShardsResponse getSearchShards(String index, ElasticsearchTableDescription tableDescription)
+    {
+        TransportClient client = clients.get(tableDescription.getClusterName());
+        verify(client != null, "TransportClient is null");
+        ClusterSearchShardsRequest request = new ClusterSearchShardsRequest(index);
+        return client.admin()
+                .cluster()
+                .searchShards(request)
+                .actionGet(timeout.toMillis());
+    }
+
+    private List<ColumnMetadata> buildMetadata(List<ElasticsearchColumn> columns)
+    {
+        List<ColumnMetadata> result = new ArrayList<>();
+        for (ElasticsearchColumn column : columns) {
+            Map<String, Object> properties = new HashMap<>();
+            properties.put("jsonPath", column.getJsonPath());
+            properties.put("jsonType", column.getJsonType());
+            properties.put("isList", column.isList());
+            properties.put("ordinalPosition", column.getOrdinalPosition());
+            result.add(new ColumnMetadata(column.getName(), column.getType(), "", "", false, properties));
+        }
+        return result;
+    }
+
+    private List<ElasticsearchColumn> buildColumns(ElasticsearchTableDescription tableDescription)
+    {
+        List<ElasticsearchColumn> columns = new ArrayList<>();
+        TransportClient client = clients.get(tableDescription.getClusterName());
+        verify(client != null, "TransportClient is null");
```

Comment:
`verify(client != null, "client is null");`

Reply:
will fix